src/edit.js

```diff
@@ -79,6 +81,62 @@ export default function Edit({attributes, setAttributes, instanceId}) {
         );
     }
 
+    const [recaptchaState, setRecaptchaState] = useState({
+        recaptchaSiteKey: '',
+        recaptchaSecretKey: '',
+    });
+
+    const siteSettings = useSelect((select) => {
+        return select('core').getEntityRecord('root', 'site');
+    }, []);
+
+    useEffect(() => {
+        if (siteSettings) {
+            const {dfb_options} = siteSettings;
+            setRecaptchaState({
+                recaptchaSiteKey: dfb_options.recaptcha_v2_site_key,
+                recaptchaSecretKey: dfb_options.recaptcha_v2_secret_key,
+            });
+            setAttributes({
+                enableRecaptcha: enableRecaptcha,
+                recaptchaSiteKey: dfb_options.recaptcha_v2_site_key,
+            });
+        }
+    }, [siteSettings]);
+
+    const submitRecaptchaCreds = () => {
+        // Admin entered a good token 👍.
+        // Save it and show a notice.
+        dispatch('core')
+            .saveEntityRecord('root', 'site', {
+                dfb_options: {
+                    recaptcha_v2_secret_key: recaptchaState.recaptchaSecretKey,
+                    recaptcha_v2_site_key: recaptchaState.recaptchaSiteKey,
+                },
+            })
+            .then((response) => {
+                dispatch('core/notices').createInfoNotice(
+                    __('🎉 Success! The API Keys have been saved.', 'donation-form-block'),
+                    {
+                        isDismissible: true,
+                        type: 'snackbar',
+                    }
+                );
+                setAttributes({
+                    recaptchaSiteKey: recaptchaState.recaptchaSiteKey,
+                    recaptchaSecretKey: recaptchaState.recaptchaSecretKey,
+                });
+            })
+            .catch((error) => {
+                console.log(error);
+                dispatch('core/notices').createNotice(error.message, {
+                    isDismissible: true,
+                    type: 'snackbar',
+                });
+                setAttributes({recaptchaSiteKey: null, recaptchaSecretKey: null});
+            });
```

Comment:
Not a big deal, but I do recommend making more use of [async/await](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function) to clean up this sort of nested, hard-to-read code.

Reply:
Don't have time to do this...